File: coding/Codeforces/ed_round_132/task_B.py
# https://codeforces.com/contest/1709/problem/B
def solving(arr, s, t):
    ans = 0
    if s < t:
        step = 1
    else:
        step = -1
    for i in range(s, t, step):
        u = arr[i] - arr[i + step]
        if u > 0:
            ans += u

    return ans

def main():
    n, m = map(int, input().split())
    arr = list(map(int, input().split()))

    left_to_right = []
    for i in range(n - 1):
        u = arr[i] - arr[i + 1]
        if u < 0:
            u = 0
        left_to_right.append(u)

    right_to_left = []
    for i in range(n - 1, 0, -1):
        u = arr[i] - arr[i - 1]
        if u < 0:
            u = 0
        right_to_left.append(u)

    right_to_left.reverse()

    answer = []
    for _ in range(m):
        s, t = map(int, input().split())
        s = s - 1
        t = t - 1
        if s < t:
            ans = sum(left_to_right[s: t])
        else:
            ans = sum(right_to_left[t: s])

        answer.append(ans)
    answer = "\n".join(map(str, answer))
    print(answer)

if __name__ == "__main__":
    main()

File: coding/Codeforces/round_929/task_B.cpp
// https://codeforces.com/contest/1933/problem/B
// Created by ruslan515 on 27.02.2024
//

#include<bits/stdc++.h>

using namespace std;

void solving() {
    int n;
    cin >> n;
    int sum = 0;
    vector<int> a(n);
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
        sum += a[i];
    }
    int d = sum % 3;
    int answer = 0;
    if (d == 1 && a.size() != 1) {
        answer = 2;
        for (int i = 0; i < n; ++i) {
            int x = (sum - a[i]) % 3;
            if (x == 0) {
                answer = 1;
                break;
            }
        }
    } else if (d == 1 && a.size() == 1) {
        answer = 1;
    } else if (d == 2) {
        answer = 1;
    }

    cout << answer << "\n";
    return;
}

int main() {
    ios::sync_with_stdio(false);
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        solving();
    }

    return 0;
}

